pkg/graveler/ref/manager.go

```diff
@@ -47,23 +47,75 @@ func NewKVRefManager(executor batch.Batcher, kvStore kv.StoreMessage, db db.Data
 }
 
 func (m *KVManager) GetRepository(ctx context.Context, repositoryID graveler.RepositoryID) (*graveler.Repository, error) {
-	return m.db.GetRepository(ctx, repositoryID)
+	repo := graveler.RepositoryData{}
+	_, err := m.kvStore.GetMsg(ctx, graveler.RepoPartition(), []byte(graveler.RepoPath(repositoryID)), &repo)
+	if err != nil {
+		if errors.Is(err, kv.ErrNotFound) {
+			err = graveler.ErrRepositoryNotFound
+		}
+		return nil, err
+	}
+	return graveler.RepoFromProto(&repo).Repository, nil
+}
+
+func (m *KVManager) createBareRepository(ctx context.Context, repositoryID graveler.RepositoryID, repository graveler.Repository) error {
+	repoRecord := &graveler.RepositoryRecord{
+		RepositoryID: repositoryID,
+		Repository:   &repository,
+	}
+	repo := graveler.ProtoFromRepo(repoRecord)
+	err := m.kvStore.SetMsgIf(ctx, graveler.RepoPartition(), []byte(graveler.RepoPath(repositoryID)), repo, nil)
+	if err != nil {
+		if errors.Is(err, kv.ErrPredicateFailed) {
+			err = graveler.ErrNotUnique
+		}
+		return err
+	}
+	return nil
 }
 
 func (m *KVManager) CreateRepository(ctx context.Context, repositoryID graveler.RepositoryID, repository graveler.Repository, token graveler.StagingToken) error {
-	return m.db.CreateRepository(ctx, repositoryID, repository, token)
+	firstCommit := graveler.NewCommit()
+	firstCommit.Message = graveler.FirstCommitMsg
+	firstCommit.Generation = 1
+	// commitID := m.addressProvider.ContentAddress(firstCommit)
```

Comment:
Why is it commented out?

Reply:
`AddCommit` does not  receive a `commitID` as a parameter, so in order to use it here, I had to remove it.
This can be returned when the `Commit` entitiy support is implemented, possibly in a similar way to what we have in the DB ref manager, where an internal `addCommit` is receiveing a commit and its ID and performs the actual DB update, while the public `AddCommit` uses it.
I will add a comment to indicate that

Reply:
Rethinking it, this `commitID` is used for the purpose of creating the default branch before the commit itself, which doesn't make any sense, right?
Am I missing something?
I'm removing this line altogether, as I switched the order of creation of `branch` and `commit`